src/main/java/com/teambash/keepinmind/service/impl/SynonymServiceImp.java

```diff
@@ -0,0 +1,48 @@
+package com.teambash.keepinmind.service.impl;
+
+import com.teambash.keepinmind.model.Synonym;
+import com.teambash.keepinmind.repository.SynonymRepository;
+import com.teambash.keepinmind.service.SynonymService;
+import com.teambash.keepinmind.shared.dto.SynonymDto;
+import lombok.extern.java.Log;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Log
+public class SynonymServiceImp implements SynonymService {
+
+  @Autowired private SynonymRepository synonymRepository;
+
+  @Autowired private ModelMapper modelMapper;
+
+  @Override
+  public SynonymDto createSynonym(SynonymDto dto) {
+    Optional<Synonym> synonymByName = synonymRepository.findByName(dto.getName());
+    if (synonymByName.isPresent()) {
+      return modelMapper.map(Synonym.builder().build(), SynonymDto.class);
```

Comment:
Could you just return an empty DTO instance?

Reply:
No, actually this is an issue I should return 

`return modelMapper.map(synonymByName.get(),SynonymDto.class);`

instead of that.